nf_core/components/install.py

```diff
@@ -50,6 +50,15 @@ def __init__(
             self.installed_by = [self.component_type]
 
     def install(self, component: str, silent: bool = False) -> bool:
+        if isinstance(component, dict):
+            if component.get("git_remote") is not None:
+                remote_url = component.get("git_remote")
+                branch = component.get("branch")
```

Comment:
My understanding of the way the dictionary is built, if there is "git_remote", then there must be "branch" too ?
```suggestion
                branch = component["branch"]
```